application/models/services/SurveyPermissions.php

```diff
@@ -183,12 +197,14 @@ public function getSurveyUserList()
         $userList = [];
 
         if ($this->userControlSameGroupPolicy) {
-            $authorizedUsersList = getUserList('onlyuidarray');
+            $authorizedUsersList = $this->getFilteredUserIdList();
         }
         $index = 0;
         foreach ($users as $user) {
             if (
-                !$this->userControlSameGroupPolicy || in_array($user->uid, $authorizedUsersList)
+                $hasUserReadPermission ||
```

Comment:
204 | ERROR | [x] Boolean operators between conditions must be at the beginning or end of the line, but not both

;)